Utils/Dataflow/016_task2es/task2es.py

```diff
@@ -183,26 +183,32 @@ def input_events(data):
     return result
 
 
-def add_chain_id(data):
-    """ Add chain_id into data - taskid of the task chain's root.
+def transform_chain_data(data):
+    """ Transform chain_data into array of integers and get chain_id from it.
+
+    chain_id is the taskid of the task chain's root.
 
     :param data: data to be updated, must contain taskid and proper chain_data
     (string of numbers separated by commas).
     :type data: dict
 
-    :return: True if update was successful, False otherwise (data is unchanged)
+    :return: True if update was successful, False otherwise (chain_data will be
+    discarded from data in such case)
     :rtype: bool
     """
     if type(data) is not dict:
         return False
     taskid = data.get('taskid')
     chain_data = data.get('chain_data')
     if not chain_data or not chain_data.replace(',', '').isdigit():
-        sys.stderr.write('(WARN) Task %s: cannot obtain chain_id, '
-                         'chain_data "%s" seems to be incorrect.\n'
+        sys.stderr.write('(WARN) Task %s: cannot transform chain_data "%s", '
+                         'it seems to be incorrect and will be discarded.\n'
                          % (taskid, chain_data))
+        del data['chain_data']
         return False
-    data['chain_id'] = int(chain_data.split(',')[0])
+    chain_data = [int(i) for i in chain_data.split(',')]
+    data['chain_id'] = chain_data[0]
+    data['chain_data'] = chain_data
```

Comment:
What do you think about additional logic, like "If there is no `chain_data`, then `chain_id` = `task_id` and `chain_data` = `[task_id]`"? We may be sure that 009 won\`t produce empty `chain_data` for now, but who said that:
* it will never change;
* this stage won\`t get data from somewhere else?

Reply:
Done, assuming that you meant "if there is no VALID `chain_data`" - `chain_data` containing a jumble of random symbols does not look any more useful to me than the field missing outright.